hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -188,40 +193,64 @@ def test_function(self, data):
                             self.database.delete(
                                 self.covering_key, existing.buffer)
 
+        # Find or create the node that would be the parent of this result,
+        # and note any forced or masked bytes along the way.
```

Comment:
I find this comment a little misleading, as it suggests it's referring to the following lines but it's actually referring to this entire process. I'm not sure what the best fix is - maybe rephrase slightly and add an extra blank line?

Reply:
The comment is describing the immediately-following loop. But adding a separate whole-process comment might make that clearer.

Reply:
I also noticed that the “parent” part is subtly wrong, so I'll change that to be more accurate.